rqt_image_view/src/rqt_image_view/image_view.cpp

```diff
@@ -237,14 +280,24 @@ void ImageView::onTopicChanged(int index)
   }
 }
 
-void ImageView::onZoom1(bool checked)
+void ImageView::resizeEvent(QResizeEvent *event)
 {
-  if (checked)
+  if( ui_.zoom_1_push_button->isChecked() == false )
   {
-    if (ui_.image_frame->getImage().isNull())
-    {
-      return;
-    }
+    preZoomSize_ = event->size();
+  }
+}
+
+void ImageView::onZoom(bool checked)
+{
+  if (ui_.image_frame->getImage().isNull())
+  {
+    return;
+  }
+    
+  if (ui_.zoom_1_push_button->isChecked())
```

Comment:
Why not continue to use `checked` as it was before? That's why the boolean is passed in the first place.